File: di_java/src/test/java/com/kmmania/hr/DAOImplHRTest.java
package com.kmmania.hr;

import com.kmmania.api.DAO;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.*;

class DAOImplHRTest {
    @InjectMocks
    private DAO daoMock;

    @BeforeEach
    void setUp() {
        daoMock = mock(DAOImplHR.class);
    }

    @Test
    @DisplayName("getData() method should retrieve data from DAO")
    void getDataShouldRetrieveDataFromDAO() {
        // Given
        when(daoMock.getData()).thenReturn(185.0);

        // When
        double result = daoMock.getData();

        // Then
        verify(daoMock, times(1)).getData();
        assertThat(result).isEqualTo(185);
    }
}

File: di_kotlin/src/test/kotlin/hr/DAOImplHRTest.kt
package hr

import com.kmmania.api.DAO
import com.kmmania.hr.DAOImplHR
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.InjectMocks
import org.mockito.Mockito.*

class DAOImplHRTest {
    @InjectMocks
    private lateinit var daoMock: DAO

    @BeforeEach
    fun setUp() {
        daoMock = mock(DAOImplHR::class.java)
    }

    @Test
    fun `getData() method should retrieve data from DAO`() {
        // Given
        `when`(daoMock.getData()).thenReturn(185.0)

        // When
        val result = daoMock.getData()

        // Then
        verify(daoMock, times(1)).getData()
        assertThat(result).isEqualTo(185.0)
    }
}